pkg/api/metrics.go

```diff
@@ -18,8 +122,95 @@ type Metrics struct {
 	ReceivedQueries     prometheus.Counter
 	FailedQueries       prometheus.Counter
 	QueryBatchDuration  prometheus.Histogram
-	CachedMetricNames   prometheus.CounterFunc
-	CachedLabels        prometheus.CounterFunc
 	InvalidReadReqs     prometheus.Counter
 	InvalidWriteReqs    prometheus.Counter
+	CachedMetricNames   prometheus.CounterFunc
+	HTTPRequestDuration *prometheus.HistogramVec
+	CachedLabels        prometheus.CounterFunc
+	MetricNamesCacheCap prometheus.GaugeFunc
+	LabelsCacheCap      prometheus.GaugeFunc
+}
+
+func InitMetrics() *Metrics {
+	if metrics != nil {
+		return metrics
+	}
+	prometheus.MustRegister(leaderGauge)
```

Comment:
`MustRegister()` takes on args of metrics. Instead of calling on separate lines repeatedly, we should pass all these in just one MustRegister. This will lessen the complexity of stack trace, where multiple function calls are a overheads.

Reply:
Like this:

https://github.com/prometheus/prometheus/blob/a282d2509971174301408c5a5f96946c478dfb0f/tsdb/db.go#L266

Reply:
Thanks for the tip, fixed!